### ISSUE TITLE
3.6 branch CI is failing since jupyter-server 2.0 release

### ISSUE THREAD

**Issue by krassowski** [jupyterlab/jupyterlab#13563]
> I don't know what to think about the CI failures. They seems unrelated and they happen also on the latest merge on https://github.com/jupyterlab/jupyterlab/pull/13543. But I'm surprised by their failure on the second attempt.

by @fcollonval in https://github.com/jupyterlab/jupyterlab/pull/13555#pullrequestreview-1210406165

> I think this might have something to do with Jupyter Server 2.0 release two days ago. I see that the failures are in server-related tests and that jupyter-server-2.0.0 does get installed, so looks pretty solid.
>
> I re-run Linux (usage, 3.10) with naive hope that 2.0.1 that was released 10 hours ago may have fixed it, but it does not.
>
> Maybe some server 2.0 compatibility thing that was not backported to 3.6.x?

by @krassowski in https://github.com/jupyterlab/jupyterlab/pull/13555#issuecomment-1343461554

One of the offenders appears to be a browser check using notebook server:

https://github.com/jupyterlab/jupyterlab/blob/f209231f9f3790645e5b09f61f4a1fa2adec4fb3/scripts/ci_script.sh#L240-L254

whereas on master it passes because... there isn't one:

https://github.com/jupyterlab/jupyterlab/blob/e658b6f28e3501968e7fe8e211677c5378be14d0/scripts/ci_script.sh#L234-L239

<details>

The relevant log is:

```python
[W 09:34:39.685 NotebookApp] Config option `nbserver_extensions` not recognized by `NotebookApp`.  Did you mean one of: `reraise_server_extension_failures, server_extensions`?
[I 09:34:39.698 NotebookApp] Writing notebook server cookie secret to /Users/runner/Library/Jupyter/runtime/notebook_cookie_secret
[I 2022-12-08 09:34:39.881 LabApp] JupyterLab extension loaded from /Users/runner/work/jupyterlab/jupyterlab/jupyterlab
[I 2022-12-08 09:34:39.881 LabApp] JupyterLab application directory is /Users/runner/hostedtoolcache/Python/3.10.8/x64/share/jupyter/lab
[I 2022-12-08 09:34:39.912 BrowserApp] JupyterLab extension loaded from /Users/runner/work/jupyterlab/jupyterlab/jupyterlab
[I 2022-12-08 09:34:39.912 BrowserApp] JupyterLab application directory is /Users/runner/hostedtoolcache/Python/3.10.8/x64/share/jupyter/lab
[I 09:34:39.918 NotebookApp] Serving notebooks from local directory: /Users/runner/work/jupyterlab/jupyterlab
[I 09:34:39.919 NotebookApp] Jupyter Notebook 6.5.2 is running at:
[I 09:34:39.919 NotebookApp] http://localhost:8888/?token=a5ce2adcfa31320d40b0214de6a784d54ab272e205d898c8
[I 09:34:39.919 NotebookApp]  or http://127.0.0.1:8888/?token=a5ce2adcfa31320d40b0214de6a784d54ab272e205d898c8
[I 09:34:39.919 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 09:34:39.926 NotebookApp] 
    
    To access the notebook, open this file in a browser:
        file:///Users/runner/Library/Jupyter/runtime/nbserver-43188-open.html
    Or copy and paste one of these URLs:
        http://localhost:8888/?token=a5ce2adcfa31320d40b0214de6a784d54ab272e205d898c8
     or http://127.0.0.1:8888/?token=a5ce2adcfa31320d40b0214de6a784d54ab272e205d898c8
[I 09:34:39.931 NotebookApp] Running async test
Starting headless webkit...
Navigating to page: file:///Users/runner/Library/Jupyter/runtime/nbserver-43188-open.html
Waiting for page to load...
/Users/runner/hostedtoolcache/Python/3.10.8/x64/lib/python3.10/site-packages/jupyter_server/base/handlers.py:232: UserWarning: The Tornado web application does not have an 'identity_provider' defined in its settings. In future releases of jupyter_server, this will be a required key for all subclasses of `JupyterHandler`. For an example, see the jupyter_server source code for how to add an identity provider to the tornado settings: https://github.com/jupyter-server/jupyter_server/blob/aa8fd8b3faf37466eeb99689d5555314c5bf6640/jupyter_server/serverapp.py#L253
  warnings.warn(
[I 09:34:43.098 NotebookApp] 302 GET /lab?token=a5ce2adcfa31320d40b0214de6a784d54ab272e205d898c8 (::1) 4.140000ms
Waiting for page content..
Error loading JupyterLab page:
Waiting for #main selector...
<!DOCTYPE html><html lang="en-us"><head>
    <meta charset="utf-8">

    <title>Jupyter Notebook</title>
```

</details>

It was removed in commit 5a95a5575a0353af88ba29de984d17357f2e6b2a and PR #11894.

- should we backport #11894?
- should we test 3.6 with both jupyter-server 1.x and 2.x?

**Comment by fcollonval** [comment]
I don't know what is the best path forward here because actually the following is already failing:

```sh
# Create a fresh environment with JupyterLab 3.5
mamba create -yn tlab35 jupyterlab
mamba activate tlab35
jupyter notebook --NotebookApp.default_url="/lab"
```

This is not working because the handler for `jupyter_server` v2 (that got installed) are not compatible with notebook v6.

Error trace:

```
[I 2023-01-04 16:38:16.028 LabApp] JupyterLab extension loaded from /home/fcollonval/miniconda3/envs/tlab35/lib/python3.11/site-packages/jupyterlab
[I 2023-01-04 16:38:16.028 LabApp] JupyterLab application directory is /home/fcollonval/miniconda3/envs/tlab35/share/jupyter/lab
[I 16:38:16.030 NotebookApp] Serving notebooks from local directory: /home/fcollonval
[I 16:38:16.030 NotebookApp] Jupyter Notebook 6.5.2 is running at:
[I 16:38:16.030 NotebookApp] http://localhost:8888/?token=5b7efaddeb631f0baa56c3ca5cc289dd4ad64c8cac0481dd
[I 16:38:16.030 NotebookApp]  or http://127.0.0.1:8888/?token=5b7efaddeb631f0baa56c3ca5cc289dd4ad64c8cac0481dd
[I 16:38:16.030 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 16:38:16.047 NotebookApp] 
    
    To access the notebook, open this file in a browser:
        file:///home/fcollonval/.local/share/jupyter/runtime/nbserver-56753-open.html
    Or copy and paste one of these URLs:
        http://localhost:8888/?token=5b7efaddeb631f0baa56c3ca5cc289dd4ad64c8cac0481dd
     or http://127.0.0.1:8888/?token=5b7efaddeb631f0baa56c3ca5cc289dd4ad64c8cac0481dd
/home/fcollonval/miniconda3/envs/tlab35/lib/python3.11/site-packages/jupyter_server/base/handlers.py:240: UserWarning: The Tornado web application does not have an 'identity_provider' defined in its settings. In future releases of jupyter_server, this will be a required key for all subclasses of `JupyterHandler`. For an example, see the jupyter_server source code for how to add an identity provider to the tornado settings: https://github.com/jupyter-server/jupyter_server/blob/aa8fd8b3faf37466eeb99689d5555314c5bf6640/jupyter_server/serverapp.py#L253
  warnings.warn(
[E 16:38:17.322 NotebookApp] Error unpacking user from cookie: Extra data: line 1 column 4 (char 3)
[W 16:38:17.323 NotebookApp] Clearing invalid/expired login cookie username-localhost-8888
[I 16:38:17.323 NotebookApp] 302 GET /lab?token=5b7efaddeb631f0baa56c3ca5cc289dd4ad64c8cac0481dd (127.0.0.1) 2.080000ms
```

**Comment by blink1073** [comment]
I think perhaps a new version of Notebook 6 needs to be released that pins to Jupyter Server <2.

**Comment by fcollonval** [comment]
> I think perhaps a new version of Notebook 6 needs to be released that pins to Jupyter Server <2.

We cannot go that path otherwise it will prevent JupyterLab 3.5 and 3.6 to install jupyter-server as `notebook < 7` is an explicit dependency.

---

Closed by #13716
